crates/widgets/src/renderer/table_row.rs

```diff
@@ -0,0 +1,61 @@
+use crate::style::table_row::StyleSheet;
+
+use crate::widget::table_row;
+use iced_graphics::{Backend, Primitive, Renderer};
+use iced_native::{Background, Color, Element, Layout, Point, Rectangle};
+
+impl<B> table_row::Renderer for Renderer<B>
+where
+    B: Backend,
+{
+    type Style = Box<dyn StyleSheet>;
+
+    fn draw<Message>(
+        &mut self,
+        defaults: &Self::Defaults,
+        layout: Layout<'_>,
+        cursor_position: Point,
+        style_sheet: &Self::Style,
+        content: &Element<'_, Message, Self>,
+        viewport: &Rectangle,
+    ) -> Self::Output {
+        let bounds = layout.bounds();
+        let is_mouse_over = bounds.contains(cursor_position);
+        let content_layout = layout.children().next().unwrap();
+
+        let style = if is_mouse_over {
+            style_sheet.hovered()
+        } else {
+            style_sheet.style()
+        };
+
+        let (content, mouse_interaction) =
+            content.draw(self, &defaults, content_layout, cursor_position, viewport);
+
+        (
+            if style.background.is_some() {
+                let background = Primitive::Quad {
+                    bounds: Rectangle {
+                        x: bounds.x + style.offset_left as f32,
```

Comment:
We shouldn't hardcode this offset to the implementation of this widget. What if we want to use this widget, but have it highlight the entire thing? We should just wrap this widget in a Row widget and apply the padding there by pushing some Space on each side of it.

Reply:
Ahh, I see we can specify it with the style, so I guess not that big of a deal. I'm ok leaving it this way.

Reply:
And when I think about it more, we would need this anyway to work well with the nested scrollable since we don't want the scrollbar to get highlighted when hovering over it, so a offset for the styling would be needed in that case

Reply:
Omg I'm retarded, disregard everything I said. This is inside the scollable, not the other way around :P